src/geb/geb.lisp

```diff
@@ -157,3 +157,49 @@ Example:
         ((distribute x y z)  (coprod (prod x y) (prod x z)))
         ((inject-left x y)   (coprod x y))
         ((inject-right x y)  (coprod x y)))))
+
+(defun gather (x y z)
+  (pair (mcase (<-left x y) (<-left x z))
+        (mcase (comp (->left y z) (<-right x y))
+               (comp (->right y z) (<-right x z)))))
+
+(defun prod-left-assoc (f)
+  (trivia:match (dom f)
+    ((prod w (prod x y))
+     (comp f (pair (comp (<-left w x) (<-left (prod w x) y))
+                   (pair (comp (<-right w x) (<-left (prod w x) y))
+                         (<-right (prod w x) y)))))
+    (_ (error "object ~A need to be of a double product type, however it is of ~A" f (dom f)))))
+
+(defun right (f)
```

Comment:
Should we call this right? We should think of a better name, my initial response of `inject-right` also seems bad, as it looks like the injection function itself. Some thought should be had on this